velox/exec/fuzzer/MemoryArbitrationFuzzer.cpp

```diff
@@ -0,0 +1,665 @@
+
+  std::vector<std::shared_ptr<connector::ConnectorSplit>> splits;
+  for (auto i = 0; i < aggregateInput.size(); ++i) {
+    const std::string filePath = fmt::format("{}/aggregate{}", tableDir, i);
+    writeToFile(filePath, aggregateInput[i], writerPool_.get());
+    splits.push_back(makeSplit(filePath));
+  }
+
+  std::vector<PlanWithSplits> plans;
+  const auto inputRowType = asRowType(aggregateInput[0]->type());
+  {
+    const auto planNodeIdGenerator =
+        std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId scanId;
+    auto plan = PlanWithSplits{
+        PlanBuilder(planNodeIdGenerator)
+            .tableScan(inputRowType)
+            .capturePlanNodeId(scanId)
+            .singleAggregation(groupingKeys, aggregates, {})
+            .planNode(),
+        {{scanId, fromConnectorSplits(splits)}}};
+    plans.push_back(std::move(plan));
+  }
+
+  {
+    // Partial -> final aggregation plan.
+    const auto planNodeIdGenerator =
+        std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId scanId;
+    auto plan = PlanWithSplits{
+        PlanBuilder(planNodeIdGenerator)
+            .tableScan(inputRowType)
+            .capturePlanNodeId(scanId)
+            .partialAggregation(groupingKeys, aggregates, {})
+            .finalAggregation()
+            .planNode(),
+        {{scanId, fromConnectorSplits(splits)}}};
+    plans.push_back(std::move(plan));
+  }
+
+  {
+    // Partial -> intermediate -> final aggregation plan.
+    const auto planNodeIdGenerator =
+        std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId scanId;
+    auto plan = PlanWithSplits{
+        PlanBuilder(planNodeIdGenerator)
+            .tableScan(inputRowType)
+            .capturePlanNodeId(scanId)
+            .partialAggregation(groupingKeys, aggregates, {})
+            .intermediateAggregation()
+            .finalAggregation()
+            .planNode(),
+        {{scanId, fromConnectorSplits(splits)}}};
+    plans.push_back(std::move(plan));
+  }
+
+  return plans;
+}
+
+void MemoryArbitrationFuzzer::verify() {
+  const auto outputDirectory = TempDirectoryPath::create();
+  const auto spillDirectory = exec::test::TempDirectoryPath::create();
+  const auto tableScanDir = exec::test::TempDirectoryPath::create();
+
+  std::vector<PlanWithSplits> plans;
+  for (const auto& plan : hashJoinPlans(tableScanDir->getPath())) {
+    plans.push_back(plan);
+  }
+  for (const auto& plan : aggregatePlans(tableScanDir->getPath())) {
+    plans.push_back(plan);
+  }
+
+  SCOPE_EXIT {
+    waitForAllTasksToBeDeleted();
+  };
+
+  const auto startTime = std::chrono::system_clock::now();
+  const auto numThreads = FLAGS_num_batches;
+  std::vector<std::thread> queryThreads;
+  queryThreads.reserve(numThreads);
+  for (int i = 0; i < numThreads; ++i) {
+    queryThreads.emplace_back([&, i]() {
+      while (true) {
+        if (static_cast<std::chrono::duration<double>>(
```

Comment:
I think we could just use a bool: std::atomic_bool stop{false} and the test thread wait for FLAGS_iteration_duration_sec and set it true. Then all the query threads stop.